tests/runtests.py

```diff
@@ -356,24 +345,27 @@ def django_tests(verbosity, interactive, failfast, keepdb, reverse,
                  timing, shuffle):
     if verbosity >= 1:
         msg = "Testing against Django installed in '%s'" % os.path.dirname(django.__file__)
-        max_parallel = default_test_processes() if parallel == 0 else parallel
-        if max_parallel > 1:
-            msg += " with up to %d processes" % max_parallel
+        if parallel > 1:
+            msg += " with up to %d processes" % parallel
         print(msg)
 
     test_labels, state = setup_run_tests(verbosity, start_at, start_after, test_labels)
     # Run the test suite, including the extra validation tests.
     if not hasattr(settings, 'TEST_RUNNER'):
         settings.TEST_RUNNER = 'django.test.runner.DiscoverRunner'
     TestRunner = get_runner(settings)
+
+    if not all(conn.features.can_clone_databases for conn in connections.all()):
+        parallel = 1
+
```

Comment:
We shouldn't silently change passed parameters. IMO it better to raise an exception like we do now:
```
$ export DJANGO_SETTINGS_MODULE=test_oracle
$ ./runtests.py queries --parallel=2
Testing against Django installed in '/django/django' with up to 2 processes
Found 416 test(s).
Creating test database for alias 'default'...
Creating test user...
Cloning test database for alias 'default'...
Traceback (most recent call last):
  File "./runtests.py", line 659, in <module>
    failures = django_tests(
  File "./runtests.py", line 385, in django_tests
    failures = test_runner.run_tests(test_labels)
  File "/django/django/test/runner.py", line 881, in run_tests
    old_config = self.setup_databases(
  File "/django/django/test/runner.py", line 787, in setup_databases
    return _setup_databases(
  File "/django/django/test/utils.py", line 217, in setup_databases
    connection.creation.clone_test_db(
  File "/django/django/db/backends/base/creation.py", line 239, in clone_test_db
    self._clone_test_db(suffix, verbosity, keepdb)
  File "/django/django/db/backends/base/creation.py", line 255, in _clone_test_db
    raise NotImplementedError(
NotImplementedError: The database backend doesn't support cloning databases. Disable the option to run tests in parallel processes.
```

Reply:
This is moved from the previous behaviour in `actual_test_processes()`

Reply:
Not exactly, we only changed the `parallel` to `1` when it was not given. Please see an example call :point_up:

Reply:
Ah right, I see, will fix.